extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/JsonRegistryProcessor.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.micrometer.deployment.export;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.micrometer.deployment.MicrometerRegistryProviderBuildItem;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+import io.quarkus.micrometer.runtime.export.JsonMeterRegistryProvider;
+import io.quarkus.micrometer.runtime.export.JsonRecorder;
+import io.quarkus.micrometer.runtime.registry.json.JsonMeterRegistry;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+
+public class JsonRegistryProcessor {
+
+    private static final Logger log = Logger.getLogger(JsonRegistryProcessor.class);
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    public void initializeJsonRegistry(MicrometerConfig config,
+            BuildProducer<MicrometerRegistryProviderBuildItem> registryProviders,
+            BuildProducer<RouteBuildItem> routes,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            JsonRecorder recorder,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        if (config.checkRegistryEnabledWithDefault(config.export.json)) {
+            additionalBeans.produce(AdditionalBeanBuildItem.builder()
+                    .addBeanClass(JsonMeterRegistryProvider.class)
+                    .setUnremovable().build());
+            registryProviders.produce(new MicrometerRegistryProviderBuildItem(JsonMeterRegistry.class));
+            routes.produce(new RouteBuildItem(recorder.route(config.export.json.path), recorder.getHandler()));
+            reflectiveClasses.produce(ReflectiveClassBuildItem
+                    .builder("org.HdrHistogram.Histogram",
+                            "org.HdrHistogram.DoubleHistogram",
+                            "org.HdrHistogram.ConcurrentHistogram")
+                    .constructors(true).build());
```

Comment:
Why is reflection necessary for these classes? I'm missing something

Reply:
I don't know, but without this, native mode breaks.